python/dialects/_AIE_util.py

```diff
@@ -0,0 +1,169 @@
+import inspect
+from functools import wraps
+
+from ..ir import *
+from ..dialects import arith
+from ..dialects import memref
+from ..dialects.func import *
+from ..dialects.scf import *
+
+
+# Create a signless arith constant of given width (default is i32).
+class integerConstant(arith.ConstantOp):
```

Comment:
Instead of having many different such specializations you're better off doing something like this https://github.com/makslevental/mlir-python-utils/blob/main/mlir/utils/dialects/ext/arith.py#L57.

Reply:
I agree. The reason why I didn't make this more complete is because I am looking forward to using your proposed method when it is available.